pyrepl/keymaps.py

```diff
@@ -18,68 +18,69 @@
-     (r'\M-\n', 'self-insert'),
-     (r'\<backslash>', 'self-insert')] + \
-    [(c, 'self-insert')
-     for c in map(chr, range(32, 127)) if c <> '\\'] + \
-    [(c, 'self-insert')
-     for c in map(chr, range(128, 256)) if c.isalpha()] + \
-    [(r'\<up>', 'up'),
-     (r'\<down>', 'down'),
-     (r'\<left>', 'left'),
-     (r'\<right>', 'right'),
-     (r'\<insert>', 'quoted-insert'),
-     (r'\<delete>', 'delete'),
-     (r'\<backspace>', 'backspace'),
-     (r'\M-\<backspace>', 'backward-kill-word'),
-     (r'\<end>', 'end'),
-     (r'\<home>', 'home'),
-     (r'\<f1>', 'help'),
-     (r'\EOF', 'end'),  # the entries in the terminfo database for xterms
-     (r'\EOH', 'home'), # seem to be wrong.  this is a less than ideal
-                        # workaround
-     ])
+    [
+        (r"\C-a", "beginning-of-line"),
+        (r"\C-b", "left"),
+        (r"\C-c", "interrupt"),
+        (r"\C-d", "delete"),
+        (r"\C-e", "end-of-line"),
+        (r"\C-f", "right"),
+        (r"\C-g", "cancel"),
+        (r"\C-h", "backspace"),
+        (r"\C-j", "self-insert"),
+        (r"\<return>", "accept"),
+        (r"\C-k", "kill-line"),
+        (r"\C-l", "clear-screen"),
+        #     (r'\C-m', 'accept'),
+        (r"\C-q", "quoted-insert"),
+        (r"\C-t", "transpose-characters"),
+        (r"\C-u", "unix-line-discard"),
+        (r"\C-v", "quoted-insert"),
+        (r"\C-w", "unix-word-rubout"),
+        (r"\C-x\C-u", "upcase-region"),
+        (r"\C-y", "yank"),
+        (r"\C-z", "suspend"),
+        (r"\M-b", "backward-word"),
+        (r"\M-c", "capitalize-word"),
+        (r"\M-d", "kill-word"),
+        (r"\M-f", "forward-word"),
+        (r"\M-l", "downcase-word"),
+        (r"\M-t", "transpose-words"),
+        (r"\M-u", "upcase-word"),
+        (r"\M-y", "yank-pop"),
+        (r"\M--", "digit-arg"),
+        (r"\M-0", "digit-arg"),
+        (r"\M-1", "digit-arg"),
+        (r"\M-2", "digit-arg"),
+        (r"\M-3", "digit-arg"),
+        (r"\M-4", "digit-arg"),
+        (r"\M-5", "digit-arg"),
+        (r"\M-6", "digit-arg"),
+        (r"\M-7", "digit-arg"),
+        (r"\M-8", "digit-arg"),
+        (r"\M-9", "digit-arg"),
+        (r"\M-\n", "self-insert"),
+        (r"\<backslash>", "self-insert"),
+    ]
+    + [(c, "self-insert") for c in map(chr, range(32, 127)) if c != "\\"]
+    + [(c, "self-insert") for c in map(chr, range(128, 256)) if c.isalpha()]
+    + [
+        (r"\<up>", "up"),
+        (r"\<down>", "down"),
+        (r"\<left>", "left"),
+        (r"\<right>", "right"),
+        (r"\<insert>", "quoted-insert"),
+        (r"\<delete>", "delete"),
+        (r"\<backspace>", "backspace"),
+        (r"\M-\<backspace>", "backward-kill-word"),
+        (r"\<end>", "end"),
+        (r"\<home>", "home"),
+        (r"\<f1>", "help"),
+        (r"\EOF", "end"),  # the entries in the terminfo database for xterms
+        (r"\EOH", "home"),  # seem to be wrong.  this is a less than ideal
+        # workaround
```

Comment:
The comment should get moved above (the end/home lines/block) maybe..?

Reply:
with some fix darker it is possible to make that as single line https://github.com/pypy/pyrepl/pull/48/commits/1582e6dab5d9bd4cd180eabb45ac1dbc3f7bbde2